rust-k256/src/lib.rs

```diff
@@ -121,59 +93,68 @@ enum PlumeVersion {
 // hash[m, gsk]^[r + sk * c] / (hash[m, pk]^sk)^c = hash[m, pk]^r
 // c = hash2(g, g^sk, hash[m, g^sk], hash[m, pk]^sk, gr, hash[m, pk]^r)
 fn verify_signals(
-    m: &[u8],
+    message: &[u8],
     pk: &ProjectivePoint,
     nullifier: &ProjectivePoint,
     c: &Output<Sha256>,
-    r_sk_c: &Scalar,
-    g_r_option: &Option<ProjectivePoint>,
-    hash_m_pk_pow_r_option: &Option<ProjectivePoint>,
+    s: &Scalar,
+    r_point_optional: &Option<ProjectivePoint>,
+    hashed_to_curve_optional: &Option<ProjectivePoint>,
     version: PlumeVersion,
 ) -> bool {
-    let mut verified: bool = true;
+    let mut verified: bool = true; // looks like antipattern to @skaunov; also see #22
```

Comment:
Oh whats the antipattern here?

Reply:
Oh, I meant #61, not #22 actually.

I perceive as anti-pattern defaulting to valid reply. "`true` until proven `false`" in case of validation doesn't seems right to me.

Reply:
Sweet! Yeah false until true makes sense, and #61 makes sense too. Let's make those changes!